### ISSUE TITLE
FOCAL FOSSA!

### ISSUE THREAD

**Issue by saltedcoffii** [dnschneid/crouton#4265]
I'm sure that all of us Linux enthusiasts are really excited about the release of **Focal Foosa**, Canonical's latest Ubuntu release, 20.04 LTS, scheduled to be released on April 23. I love and adore crouton, but as time passes, Xenial becomes more and more dated. Xenial repositories aren't just keeping up with everything new, like the latest versions of `java`, `firefox`, `mono`, `gnome`, `gcc`, *et cetera*. When (or if, I suppose,) will crouton be updating to a newer version of Ubuntu? What is in the works already? When will `crouton -r help` show `focal*` after `eoan*`?

**Comment by justchen1369** [comment]
18.04 has been avaliable for a while now.

**Comment by kingliamb** [comment]
It has been available, but it's not officially supported by crouton, and it's much too buggy productivity use.

**Comment by dnschneid** [comment]
I've added focal* to the list.

We have pretty limited time to go through and repair all the compatibility issues that appear in each release, so we've been sticking to every-other LTS as the default crouton one and try to make that work as well as we can.  Focal would be the next on the list, but it'll come down to when @drinkcat and I can allocate a week to hash it out.  Rest assured it'll be sometime before April 2021 :)

Patches welcome, as always.

**Comment by jvschiavo** [comment]
> I've added focal* to the list.
> 
> We have pretty limited time to go through and repair all the compatibility issues that appear in each release, so we've been sticking to every-other LTS as the default crouton one and try to make that work as well as we can. Focal would be the next on the list, but it'll come down to when @drinkcat and I can allocate a week to hash it out. Rest assured it'll be sometime before April 2021 :)
> 
> Patches welcome, as always.

Why Bionic was never tagged as supported? The idea was always to skip 16.04 directly to 20.04?

Is 18.04 safe to install or upgrade on my chroot?

**Comment by hacker1024** [comment]
I just tried installing Focal, but I was met with a `sed` error. Eoan works just fine.

```
$ sudo crouton -r focal -t xorg,extension,audio,touch,kde-desktop

WARNING: focal is an unsupported release.
You will likely run into issues, but things may work with some effort.
Press Ctrl-C to abort; installation will continue in 5 seconds.
WARNING: Your rootfs is writable. Signed boot verification cannot be enabled.
If this is a surprise to you, you should do a full system recovery via USB.
Installing focal-amd64 chroot to /usr/local/chroots/focal
sed: couldn't edit /tmp/crouton.kwG/scripts/focal: not a regular file
```

**Comment by dnschneid** [comment]
Looks like I forgot to update the builder with a version of debootstrap that supports focal.  That's fixed now.

**Comment by jvschiavo** [comment]
I think this should be a pinned issue. It is the most hyped thing for crouton right now. There're lots of expectations about this.

**Comment by dnschneid** [comment]
Good idea.

**Comment by neogeo71** [comment]
> It has been available, but it's not officially supported by crouton, and it's much too buggy for productivity use with crouton.
I disagree. 18.04 LTS It runs perfectly fine in Crouton, been using it for 2 years. never had a issue.

**Comment by kingliamb** [comment]
> > It has been available, but it's not officially supported by crouton, and it's much too buggy for productivity use with crouton.
> I disagree. 18.04 LTS It runs perfectly fine in Crouton, been using it for 2 years. never had a issue.
> 

Oh it won't even launch xorg for me.

**Comment by neogeo71** [comment]
I posted my process of installing on Reddit. I will find and list it
here...

**Comment by jvschiavo** [comment]
> Oh it won't even launch xorg for me.

I updated from xenial to bionic and it is working kind of okay for me. But Alt + Backspace is not result in Delete and I can't find a way to add that back. :/

**Comment by neogeo71** [comment]
Used this successfully....



Download Crouton

cd ~/Downloads

wget "https://github.com/dnschneid/crouton/blob/master/installer/crouton"

Install Ubuntu 18.04 Bionic

sudo sh ~/Downloads/crouton -r bionic -t gnome

Fix Localization for things like gnome-terminal:

sudo enter-chroot

sudo locale-gen en_US en_US.UTF-8

export LANG=en_US.UTF-8

export LC_ALL=en_US.UTF-8

sudo dpkg-reconfigure locales

Fix mouse lag

sudo apt install xserver-xorg-input-synaptics

Fix UI lag

sudo enter-chroot

cd /usr/share/X11/xorg.conf.d

sudo ln -s /etc/crouton/xorg-intel-sna.conf 20-crouton-intel-sna.conf

Install a bunch of other shit:

sudo apt install gnome-terminal nano ttf-ubuntu-font-family gnome-calculator gnome-calendar gnome-clocks file-roller gedit bash-completion ubuntu-restricted-extras gnome-tweaks iptables darktable rawtherapee steam swell-foop eog



edit Step 2 is currently broken. I'm not sure if this fix will introduce errors, but I did get it to install:

sudo sh ~/Downloads/crouton -r bionic -t core

sudo enter-chroot

Sudo apt install xserver-xorg-video-all xserver-xorg xserver-xorg-video-intel

exit

sudo sh ~/Downloads/crouton -r bionic -t gnome -u

then continue w/ the rest of the guide.

**Comment by derpyegplant** [comment]
Focal Fossa seems to only work with xfce. it wont open in gnome
or any other desktop

**Comment by neogeo71** [comment]
> Focal Fossa seems to only work with xfce. it wont open in gnome
> or any other desktop
It works! I am running it now. Gnome works. Follow directions above.

**Comment by Miustone** [comment]
I tried everything possible on My C340-15 to get 20.04 but no success (Running Bionic rn). Will try later again! Thank You very much for this! <3

Random Question. Will it be possible to compile Android Systems on 18.04 or 20.04? I'm not sure about this since i'm running a bit buggy with 18.04 and wonder if 20.04 will allow me to do this...

**Comment by saltedcoffeeuwu** [comment]
> I tried everything possible on My C340-15 to get 20.04 but no success (Running Bionic rn). Will try later again! Thank You very much for this! <3
> 
> Random Question. Will it be possible to compile Android Systems on 18.04 or 20.04? I'm not sure about this since i'm running a bit buggy with 18.04 and wonder if 20.04 will allow me to do this...

You could ask on the subreddit

**Comment by Miustone** [comment]
> > I tried everything possible on My C340-15 to get 20.04 but no success (Running Bionic rn). Will try later again! Thank You very much for this! <3
> > Random Question. Will it be possible to compile Android Systems on 18.04 or 20.04? I'm not sure about this since i'm running a bit buggy with 18.04 and wonder if 20.04 will allow me to do this...
> 
> @Miustone You could ask on the subreddit

I don't have a reddit Account but thanks...

**Comment by TheRealFanjin** [comment]
Hey guys I just wanted to let u guys know I got focal with xfce working perfectly. No bugs, nothing. It is perfect the first time I installed it. First step towards FOCAL FOOSA!
![focalfoosa](https://user-images.githubusercontent.com/58857369/87238926-9644ec00-c3ce-11ea-820f-d5e2e8a9aa53.png)

**Comment by MisterTrash-77** [comment]
Hey, sorry for the question but is anyone able to run focal or eoan ? Focal just show a blackscreen when running "sudo startxfce4" and Eoan just ask me to reboot for dbus to install (When I do it just ask me once more to reboot for dbus).
Thanks

**Comment by pairwiserr** [comment]
> > It works! I am running it now. Gnome works. Follow directions above.

@derpyegplant was referring to Focal Fossa, but your instructions are for bionic, @neogeo71 .

**Comment by pairwiserr** [comment]
> Hey, sorry for the question but is anyone able to run focal or eoan ? 

I've tried installing `focal` with `gnome` but it automatically unmounts.


Perhaps the following log is useful:


```
method return time=1595626017.745113 sender=:1.25 -> destination=:1.125 serial=910 reply_serial=2
WARNING: Kernal has no file descriptor comparison support: Functionnot implemented
(II) modeset(0): Initializing kms color map for dept 24, 8 bpc/
freon: yes
..
...
Running exit commands...
/usr/bin/xinit: connection to X server lost
```

Is this some type of display scaling issue? I tried symlinking via

```
cd /usr/share/X11/xorg.conf.d
sudo ln -s /etc/crouton/xorg-intel-sna.conf 20-crouton-intel-sna.conf
```

And although I get a slight glitch in the display before unmounting, I'm still having no luck tracking what the cause of the issue.

**Comment by TeaWolf** [comment]
I'm running focal at the moment and everything works fine except sound and some performance issues. After trying the sound solution from [here](https://askubuntu.com/questions/325418/no-sound-in-crouton#336091) I can get sound from vlc, but still no sound from Firefox. Concerning performance I keep seeing chrome running at 20% cpu load on top. Is it increase it's niceness from -8 ? Or will this not help ?

**Comment by TheRealFanjin** [comment]
@TeaWolf Yes, I remember having the exact same audio problem. I think I solved it by doing `pulseaudio -D` or `pulseaudio --start`, I forgot which one it was.
As for the Chrome issue, I just don't use Chrome in Linux. I mostly use Vivaldi and Firefox

**Comment by satmandu** [comment]
This let ```sudo startgnome``` work for me:
```
sudo enter-chroot
sudo add-apt-repository ppa:oibaf/graphics-drivers
sudo apt dist-upgrade -y
echo MESA_LOADER_DRIVER_OVERRIDE=i965 | sudo tee -a /etc/environment
cd /usr/share/X11/xorg.conf.d && sudo ln -s /etc/crouton/xorg-intel-sna.conf 20-crouton-intel-sna.conf
```

**Comment by TeaWolf** [comment]
> @TeaWolf Yes, I remember having the exact same audio problem. I think I solved it by doing `pulseaudio -D` or `pulseaudio --start`, I forgot which one it was.
> As for the Chrome issue, I just don't use Chrome in Linux. I mostly use Vivaldi and Firefox

Thanks for the pulseaudio --start tip. that did the trick. As to the browser I also use only Firefox. I think the chrome is the one running in ChromeOS when I start the crosh shell to boot up my chroot. Is there a way to start the chroot without opening chrome?

**Comment by TheRealFanjin** [comment]
@TeaWolf The "Chrome" that is using a lot of CPU is not just the Chrome in ChromeOS, it's the whole ChromeOS system. If you close Chrome in ChromeOS, the "Chrome" in your task manager will keep running. If you're interested, there's a way to [https://github.com/dnschneid/crouton/wiki/Autostart-crouton-chroot-at-ChromeOS-startup](autostart a chroot), but it's a bit complicated and doesn't really resolve the chrome issue.
Anyway, just don't open any tabs in Chrome to minimize cpu usage and definitely close android apps, they also use a lot of CPU.

**Comment by explodersname** [comment]
Hello, Could someone please post a complete installation guide for focal here? I have looked through this thread and it seems very long and complicated to go through. I have seen several comments solving issues such as using gnome, and solving audio issues and would like it all to be combined together into one, complete tutorial for me, and many others to follow. I don't care if the desktop environment is gnome although that would be nice.

**Comment by TheRealFanjin** [comment]
@explodersname You should try installing first and see what errors/problems you get.

**Comment by explodersname** [comment]
@TheRealFanjin Already have, and havent gotten any desktop environments to work. I have tried using xinit to start gnome via .xinitrc that didnt work, and sudo startgnome shows my cursor twice, then it disapears and takes me back to chrome os with the usual "lost connection to x server" that you get when stopping xorg manually. <- the same happens with xfce.

**Comment by explodersname** [comment]
Update: Using the Targets: xfce,xorg,audio,core,x11,gtk-extra I have gotten focal fossa to work. The biggest issue I'm having is audio though. I have tried using this guide: https://github.com/dnschneid/crouton/wiki/Access-audio-hardware-directly-(ALSA,-JACK) to give direct access to the soundcard, I will keep you updated if I manage to figure out how to do this.

**Comment by explodersname** [comment]
UPDATE: FINALLY (took me literally 3 minutes to get audio setup using that guide: My complete tutorial:
Install focal Fossa with this command: sudo crouton -r focal -t xorg,audio,core,gtk-extra,x11,xfce. Wait for crouton to do its thing. Create a username and pass, then sudo enter-chroot and enter the following command: sudo usermod -a -G hwaudio "$USER" This command adds you to the hardware audio group. exit the chroot, then on chrome os enter the command: sudo initctl stop cras: this stops cras (the default audio server crouton uses, which doesnt work so is useless). Then sudo enter-chroot again and enter the command, cat /proc/asound/cards. This will list off your audio cards on your chromebook, my cards name is chtmax98090 but yours may be different. The name of your card will be the second line after "1 [" e.g. mine was 1 [chtmax98090 now enter the command nano ~/.asoundrc and paste the following: `# ~/.asoundrc
# overrides default alsa settings
# To revert back to CRAS, change "type hw" with "type cras". You do not have to change "card".

pcm.!default { 
    type hw
    card bdwrt5677
}

ctl.!default { 
    type hw
    card bdwrt5677
}` 
Nano may not be installed so install it using sudo apt install nano -y (the -y flag skips the yes or no confirmation with apt) once you've pasted that into ~/.asoundrc use the keybind ctrl+o to save and press enter, then ctrl+x to exit. next install gnome control center if not already installed using sudo apt install gnome-control-center then when its installed exit the chroot and start xfce using sudo startxfce4 and when your in press alt+f2 (arrow forward key on most chromebooks) and type in gnome-control-center and press enter, you will see the control center pop up click on audio and output device should be set to whatever you'd like to use, if you have headphones plugged in then select headphones, if not press speakers. If you'd like to plug in a new audio device you must go back to chrome os before doing so, just use the keybind ctrl+alt+shift+arrow key back, now insert your headphones etc. you can go back to focal using the keybind ctrl+alt+shift+forward arrow. I hope this helped everyone! I spent hours troubleshooting and creating this guide!

**Comment by explodersname** [comment]
> > > It works! I am running it now. Gnome works. Follow directions above.
> 
> @derpyegplant was referring to Focal Fossa, but your instructions are for bionic, @neogeo71 .

I haven't got gnome but I have made a complete guide that makes audio work aswell. @pairwiserr

**Comment by kathelynn** [comment]
> This let `sudo startgnome` work for me:
> 
> ```
> sudo enter-chroot
> sudo add-apt-repository ppa:oibaf/graphics-drivers
> sudo apt dist-upgrade -y
> echo MESA_LOADER_DRIVER_OVERRIDE=i965 | sudo tee -a /etc/environment
> cd /usr/share/X11/xorg.conf.d && sudo ln -s /etc/crouton/xorg-intel-sna.conf 20-crouton-intel-sna.conf
> ```

I get this instead:
```
error: Kernel is too old for Iris. Consider upgrading to kernel v4.16.

(II) modeset(0): Initializing kms color map for depth 24, 8 bpc.
crouton: version 1-20200724165050~master:08dac970
release: focal
architecture: amd64
xmethod: xorg
targets: x11,xorg,unity-desktop,core
host: version 13421.23.0 (Official Build) beta-channel nami 
kernel: Linux localhost 4.4.232-19101-g619abfde5ea9 #1 SMP PREEMPT Mon Sep 7 01:24:35 PDT 2020 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
Running exit commands...
/usr/bin/xinit: connection to X server lost

waiting for X server to shut down Hangup
Error org.freedesktop.DBus.Error.ServiceUnknown: The name org.chromium.LibCrosService was not provided by any .service files
method return time=1599709895.274034 sender=:1.139 -> destination=:1.181 serial=5102 reply_serial=2
   boolean true
```
here's my `sudo edit-chroot -all`:
```
name: focal
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/focal...
crouton: version 1-20200724165050~master:08dac970
release: focal
architecture: amd64
xmethod: xorg
targets: x11,xorg,unity-desktop,core
host: version 13421.23.0 (Official Build) beta-channel nami 
kernel: Linux localhost 4.4.232-19101-g619abfde5ea9 #1 SMP PREEMPT Mon Sep 7 01:24:35 PDT 2020 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
Not unmounting /mnt/stateful_partition/crouton/chroots/focal as another instance is using it.
```

**Comment by kathelynn** [comment]
Focal Foosa doesn't work for me. It does boot, but I was stuck without any display output and the way to go back to Chrome OS was to go in and out of Developer Console by doing CTRL+ALT+FORWARD then CTRL+ALT+BACKWARD.
[https://hastebin.com/rereriguxa.sql](https://hastebin.com/rereriguxa.sql)

I was able to fix it by doing the following:
> This let `sudo startgnome` work for me:
> 
> ```
> sudo enter-chroot
> sudo add-apt-repository ppa:oibaf/graphics-drivers
> sudo apt dist-upgrade -y
> echo MESA_LOADER_DRIVER_OVERRIDE=i965 | sudo tee -a /etc/environment
> cd /usr/share/X11/xorg.conf.d && sudo ln -s /etc/crouton/xorg-intel-sna.conf 20-crouton-intel-sna.conf
> ```
This time, the display output works and I could see my mouse pointer moving for a split second until it crashes with an error this time:
`error: Kernel is too old for Iris. Consider upgrading to kernel v4.16.`
[https://hastebin.com/sakogujulu.sql](https://hastebin.com/sakogujulu.sql)

I reinstalled Focal Foosa as the realization hit me that Unity is not used on newer Ubuntu releases anymore, hoping that something would be fixed. Same issues, so I attempted to leave out the line to echo override on /etc/environment. Here's what I did:
```
 sudo enter-chroot
 sudo add-apt-repository ppa:oibaf/graphics-drivers
 sudo apt dist-upgrade -y
 cd /usr/share/X11/xorg.conf.d && sudo ln -s /etc/crouton/xorg-intel-sna.conf 20-crouton-intel-sna.conf
 ```
To end up with a new error.
```
xf86EnableIOPorts: failed to set IOPL for I/O (Operation not permitted)
pkexec version 0.105
Error executing command as another user: Not authorized

This incident has been reported.
crouton: version 1-20200724165050~master:08dac970
release: focal
architecture: amd64
xmethod: xorg
targets: gnome-desktop,xorg,x11,gtk-extra,cli-extra
host: version 13421.23.0 (Official Build) beta-channel nami 
kernel: Linux localhost 4.4.232-19101-g619abfde5ea9 #1 SMP PREEMPT Mon Sep 7 01:24:35 PDT 2020 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
xrandr: Output eDP1 is not disconnected but has no modes
xrandr: Output DP1 is not disconnected but has no modes
xrandr: Output DP2 is not disconnected but has no modes
xrandr: Output HDMI1 is not disconnected but has no modes
xrandr: Output HDMI2 is not disconnected but has no modes
Kicking xrandr again
xrandr: Output eDP1 is not disconnected but has no modes
xrandr: Output DP1 is not disconnected but has no modes
xrandr: Output DP2 is not disconnected but has no modes
xrandr: Output HDMI1 is not disconnected but has no modes
xrandr: Output HDMI2 is not disconnected but has no modes
Kicking xrandr again
Running exit commands...
/usr/bin/xinit: connection to X server lost
```
[https://hastebin.com/vaguvotizu.sql](https://hastebin.com/vaguvotizu.sql)

I'm still trying to make it work but I'm having tough luck here.

**Comment by clvgt12** [comment]
BTW, this rocks.

I set up my HP Chromebook x360 with Focal Fossa, and added in Sommelier/Wayland support using this [post](https://github.com/jcdang/chromeos-ubuntu-sommelier#readme) as a guide.

I can now export Firefox and Thunderbird displays back towards my Chromebook's native Wayland Display Server.  I used this [post](https://blog.szypowi.cz/en/post/ubuntu-wayland/) as reference for the Mozilla apps.

**Comment by satmandu** [comment]
@clvgt12 FYI we have the current version of sommelier working in chromebrew.

**Comment by ayr-ton** [comment]
Is bionic stable enough?

**Comment by adamallgood** [comment]
I got sound via the speakers to work in my focal fossa chroot using @explodersname 's [instructions above](https://github.com/dnschneid/crouton/issues/4265#issuecomment-689346174); however, I cannot get sound to switch over to headphones. When I open the Volume Control, the device is set to kbida7219m98927 Stereo and the Port says Headphones with no other options in the pulldown.

Has anyone had success getting both speaker sound and headphone sound working? Both still work in ChromeOS, but sometimes plugging/unplugging the headphones causes sound to stop working in the chroot. Sound even stopped working on the ChromeOS side at one point, and sudo stop cras/sudo start cras fixed it. Rebooting also fixes it.

**Comment by quickreactor** [comment]
> I got sound via the speakers to work in my focal fossa chroot using @explodersname 's [instructions above](https://github.com/dnschneid/crouton/issues/4265#issuecomment-689346174); however, I cannot get sound to switch over to headphones. When I open the Volume Control, the device is set to kbida7219m98927 Stereo and the Port says Headphones with no other options in the pulldown.
> 
> Has anyone had success getting both speaker sound and headphone sound working? Both still work in ChromeOS, but sometimes plugging/unplugging the headphones causes sound to stop working in the chroot. Sound even stopped working on the ChromeOS side at one point, and sudo stop cras/sudo start cras fixed it. Rebooting also fixes it.

When I click on the hastebin link in the guide above there is nothing there! Can you help me find the original text?

**Comment by explodersname** [comment]
I gotchu fam: 

# ~/.asoundrc
# overrides default alsa settings
# To revert back to CRAS, change "type hw" with "type cras". You do not have to change "card".

pcm.!default { 
    type hw
    card bdwrt5677
}

ctl.!default { 
    type hw
    card bdwrt5677
}

**Comment by aghebert** [comment]
> I got sound via the speakers to work in my focal fossa chroot using @explodersname 's [instructions above](https://github.com/dnschneid/crouton/issues/4265#issuecomment-689346174); however, I cannot get sound to switch over to headphones. When I open the Volume Control, the device is set to kbida7219m98927 Stereo and the Port says Headphones with no other options in the pulldown.
> 
> Has anyone had success getting both speaker sound and headphone sound working? Both still work in ChromeOS, but sometimes plugging/unplugging the headphones causes sound to stop working in the chroot. Sound even stopped working on the ChromeOS side at one point, and sudo stop cras/sudo start cras fixed it. Rebooting also fixes it.

I'm experiencing this as well.

**Comment by adamallgood** [comment]
> I'm experiencing this as well.

I have a 3.5mm jack to USB-C adaptor, and found that my headset works fine when plugged into my USB-C port. So it seems that it's sound through the 3.5mm port that does not work.

**Comment by jacob1298** [comment]
1 more day and focal being supported with crouton will be late...

**Comment by danieltahara** [comment]
> 1 more day and focal being supported with crouton will be late...

Remember, this is an open source project supported by volunteers who are giving us a really useful thing. It's not like dnschneid and co don't know when 16.04 LTS ends/ended.

Am I bummed Focal isn't officially supported yet? Of course. But making a passive aggressive comment about it and not being willing to chip in isn't charitable.

**Comment by brendan-mccoy** [comment]
Somebody's humor detector is broken.

**Comment by jacob1298** [comment]
At least I got it to work with xfce

**Comment by jacob1298** [comment]
Xfce is the only gui I know can survive a reboot

**Comment by ty1118** [comment]
kde-runtime not found. are yall going to install it

**Comment by saltedcoffii** [comment]
Today is National Crouton Day in the United States 😂

**Comment by foxsouns** [comment]
> Xfce is the only gui I know can survive a reboot

XFCE4 is _literally_ the only DE I care about. Gnome is too chonky for any of my computers, and LXDM/LXQt are lackluster & don't provide any real advantages over XFCE IMO. If I can get my Celes unbricked, I'll gladly slap Focal with XFCE on it.

**Comment by saltedcoffii** [comment]
@explodersname unfortunately the hastebin link disappeared :(
Could you post it again in another pastebin or on github?

**Comment by Snowflake6** [comment]
> UPDATE: FINALLY (took me literally 3 minutes to get audio setup using that guide: My complete tutorial:
> Install focal Fossa with this command: sudo crouton -r focal -t xorg,audio,core,gtk-extra,x11,xfce. Wait for crouton to do its thing. Create a username and pass, then sudo enter-chroot and enter the following commands:
> ...

Fresh install of Chrome OS on an Acer C720P.  The exact command above, dies with the CRAS errors reported elsewhere.  Doesn't get to creating a username and password.

**Comment by jvschiavo** [comment]
So what is the most stable install we can do right now (something that just works and doesn't require work-arounds)? Which version of Ubuntu? With which DE? Or maybe Debian?

**Comment by MrHallway0** [comment]
@jvschiavo 
@jacob1298 said:
> Xfce is the only gui I know can survive a reboot.

**Comment by Snowflake6** [comment]
> Xfce is the only gui I know can survive a reboot.

Sadly, even trying a -t xfce install (which defaults to xenial underneath) resulted in the same CRAS error.

**Comment by MrHallway0** [comment]
@Snowflake6 
> > Xfce is the only gui I know can survive a reboot.
> 
> Sadly, even trying a -t xfce install (which defaults to xenial underneath) resulted in the same CRAS error.

Strange... I haven't had any of those errors.

**Comment by satmandu** [comment]
For what it is worth, we have CRAS working with pulseaudio on  Chromebrew, so feel free to look at our setup and take what works to fix Crouton:

https://github.com/skycocker/chromebrew/blob/master/packages/cras.rb

https://github.com/skycocker/chromebrew/blob/master/packages/pulseaudio.rb

**Comment by Snowflake6** [comment]
@satmandu the problem is [known](https://github.com/dnschneid/crouton/issues/4450) and a solution exists but it hasn't been patched into the repo yet it looks like.

**Comment by satmandu** [comment]
Ah, so then it looks like this issue should be addressed once https://github.com/dnschneid/crouton/pull/4507 is merged...

**Comment by yoiammaac** [comment]
> I've added focal* to the list.
> 
> We have pretty limited time to go through and repair all the compatibility issues that appear in each release, so we've been sticking to every-other LTS as the default crouton one and try to make that work as well as we can. Focal would be the next on the list, but it'll come down to when @drinkcat and I can allocate a week to hash it out. Rest assured it'll be sometime before April 2021 :)
> 
> Patches welcome, as always.

it's july now.

**Comment by DennisLfromGA** [comment]
@yoiammaac,

> it's july now:Rest assured it'll be sometime before April **202(n)**

**Comment by yoiammaac** [comment]
> @yoiammaac,
> 
> > it's july now:Rest assured it'll be sometime before April **202(n)** :)

I Hope its soon though, or perhaps we will be continuing this with the next lts too.

**Comment by DennisLfromGA** [comment]
@yoiammaac,

```
$ edit-chroot -al
----------------------------------------------------------------------
chroot name: bionic
crouton: version 1-20210709170820~master:14ca6bc7
release: bionic
architecture: amd64
xmethod: xiwi
targets: xiwi,xorg,cli-extra,keyboard,touch,gnome-desktop,lxde-desktop,chrome-beta
host: version 13982.39.0 (Official Build) beta-channel eve 
kernel: Linux localhost 5.4.119-14931-g1ec34905b11c #1 SMP PREEMPT Wed Jun 30 20:51:02 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
----------------------------------------------------------------------
chroot name: focal
crouton: version 1-20210709170820~master:14ca6bc7
release: focal
architecture: amd64
xmethod: xiwi
targets: xiwi,xorg,cli-extra,keyboard,touch,gnome-desktop,lxde-desktop,chrome-beta
host: version 13982.39.0 (Official Build) beta-channel eve 
kernel: Linux localhost 5.4.119-14931-g1ec34905b11c #1 SMP PREEMPT Wed Jun 30 20:51:02 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
----------------------------------------------------------------------
chroot name: xenial
crouton: version 1-20210709170820~master:14ca6bc7
release: xenial
architecture: amd64
xmethod: xiwi
targets: xiwi,xorg,gnome-desktop,lxde-desktop,unity-desktop,xfce-desktop,cli-extra,keyboard,touch,chrome-beta
host: version 13982.39.0 (Official Build) beta-channel eve 
kernel: Linux localhost 5.4.119-14931-g1ec34905b11c #1 SMP PREEMPT Wed Jun 30 20:51:02 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
----------------------------------------------------------------------```

I have 3 chroots, including focal, all running and working fine with the latest crouton update/patches.

*Hope this helps,*
-**DennisLfromGA**

**Comment by dnschneid** [comment]
Huh, is it April already? :sweat_smile: 

Unfortunately, a bunch of factors (including covid) have wrecked our testing infrastructure, so this upgrade will very much be a manual (read: even more time-consuming) process.  I'm honestly not sure when @drinkcat and I will have time to churn through it.

Here's what needs to happen before we can move to focal as the default release:

1. Identify a minimal number of systems to test on.  This is a combination of kernel versions (including those from devices that are EOL!) and architectures (x86_64 and arm).  I haven't dug into what the full range of devices would look like.  Ideally we don't test more than one device on each kernel/arch combo.
2. On each of the systems, either run the crouton test suite against just focal, or manually go through each DE target (in separate chroots) and confirm they launch properly in both xiwi and xorg.  This includes testing (3D) graphics and audio in all configurations.
3. For any bug or extraneous error messages discovered, create an issue with the configuration and triage.
4. Code and test any migration needed for changing the default release.  There were some special codepaths as part of the trusty->xenial switch; they should still be in the codebase and can be extended for focal.
5. With the testing done, the migration code written, the default updated, and focal marked as supported...push the crouton release!

At the current moment, I don't have time to even oversee this process, so before everyone pipes in volunteering to help test (you are all individually and collectively the best!), we'll need someone to step forward to project manage.

**Comment by yoiammaac** [comment]
> @yoiammaac,
> 
> ```
> $ edit-chroot -al
> ----------------------------------------------------------------------
> chroot name: bionic
> crouton: version 1-20210709170820~master:14ca6bc7
> release: bionic
> architecture: amd64
> xmethod: xiwi
> targets: xiwi,xorg,cli-extra,keyboard,touch,gnome-desktop,lxde-desktop,chrome-beta
> host: version 13982.39.0 (Official Build) beta-channel eve 
> kernel: Linux localhost 5.4.119-14931-g1ec34905b11c #1 SMP PREEMPT Wed Jun 30 20:51:02 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
> freon: yes
> ----------------------------------------------------------------------
> chroot name: focal
> crouton: version 1-20210709170820~master:14ca6bc7
> release: focal
> architecture: amd64
> xmethod: xiwi
> targets: xiwi,xorg,cli-extra,keyboard,touch,gnome-desktop,lxde-desktop,chrome-beta
> host: version 13982.39.0 (Official Build) beta-channel eve 
> kernel: Linux localhost 5.4.119-14931-g1ec34905b11c #1 SMP PREEMPT Wed Jun 30 20:51:02 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
> freon: yes
> ----------------------------------------------------------------------
> chroot name: xenial
> crouton: version 1-20210709170820~master:14ca6bc7
> release: xenial
> architecture: amd64
> xmethod: xiwi
> targets: xiwi,xorg,gnome-desktop,lxde-desktop,unity-desktop,xfce-desktop,cli-extra,keyboard,touch,chrome-beta
> host: version 13982.39.0 (Official Build) beta-channel eve 
> kernel: Linux localhost 5.4.119-14931-g1ec34905b11c #1 SMP PREEMPT Wed Jun 30 20:51:02 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
> freon: yes
> ----------------------------------------------------------------------```
> ```
> 
> I have 3 chroots, including focal, all running and working fine with the latest crouton update/patches.
> 
> _Hope this helps,_
> -**DennisLfromGA**

Only xenial works for me focal shows a blank screen and cursor then back to my shell.

**Comment by yoiammaac** [comment]
> Huh, is it April already? sweat_smile
> 
> Unfortunately, a bunch of factors (including covid) have wrecked our testing infrastructure, so this upgrade will very much be a manual (read: even more time-consuming) process. I'm honestly not sure when @drinkcat and I will have time to churn through it.
> 
> Here's what needs to happen before we can move to focal as the default release:
> 
>     1. Identify a minimal number of systems to test on.  This is a combination of kernel versions (including those from devices that are EOL!) and architectures (x86_64 and arm).  I haven't dug into what the full range of devices would look like.  Ideally we don't test more than one device on each kernel/arch combo.
> 
>     2. On each of the systems, either run the crouton test suite against just focal, or manually go through each DE target (in separate chroots) and confirm they launch properly in both xiwi and xorg.  This includes testing (3D) graphics and audio in all configurations.
> 
>     3. For any bug or extraneous error messages discovered, create an issue with the configuration and triage.
> 
>     4. Code and test any migration needed for changing the default release.  There were some special codepaths as part of the trusty->xenial switch; they should still be in the codebase and can be extended for focal.
> 
>     5. With the testing done, the migration code written, the default updated, and focal marked as supported...push the crouton release!
> 
> 
> At the current moment, I don't have time to even oversee this process, so before everyone pipes in volunteering to help test (you are all individually and collectively the best!), we'll need someone to step forward to project manage.

Ok, I understand its very difficult, but just saying that xenial is a thing of yesterday now.

**Comment by BurdonSanderson** [comment]
> @yoiammaac,
> 
> ```
> $ edit-chroot -al
> ----------------------------------------------------------------------
> chroot name: bionic
> crouton: version 1-20210709170820~master:14ca6bc7
> release: bionic
> architecture: amd64
> xmethod: xiwi
> targets: xiwi,xorg,cli-extra,keyboard,touch,gnome-desktop,lxde-desktop,chrome-beta
> host: version 13982.39.0 (Official Build) beta-channel eve 
> kernel: Linux localhost 5.4.119-14931-g1ec34905b11c #1 SMP PREEMPT Wed Jun 30 20:51:02 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
> freon: yes
> ----------------------------------------------------------------------
> chroot name: focal
> crouton: version 1-20210709170820~master:14ca6bc7
> release: focal
> architecture: amd64
> xmethod: xiwi
> targets: xiwi,xorg,cli-extra,keyboard,touch,gnome-desktop,lxde-desktop,chrome-beta
> host: version 13982.39.0 (Official Build) beta-channel eve 
> kernel: Linux localhost 5.4.119-14931-g1ec34905b11c #1 SMP PREEMPT Wed Jun 30 20:51:02 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
> freon: yes
> ----------------------------------------------------------------------
> chroot name: xenial
> crouton: version 1-20210709170820~master:14ca6bc7
> release: xenial
> architecture: amd64
> xmethod: xiwi
> targets: xiwi,xorg,gnome-desktop,lxde-desktop,unity-desktop,xfce-desktop,cli-extra,keyboard,touch,chrome-beta
> host: version 13982.39.0 (Official Build) beta-channel eve 
> kernel: Linux localhost 5.4.119-14931-g1ec34905b11c #1 SMP PREEMPT Wed Jun 30 20:51:02 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
> freon: yes
> ----------------------------------------------------------------------```
> ```
> 
> I have 3 chroots, including focal, all running and working fine with the latest crouton update/patches.
> 
> _Hope this helps,_
> -**DennisLfromGA**

I'm curious if the kernel is a first order issue here.

I'm struggling to get a functioning install of focal xfce.

The first error message I get is

`error: Kernel is too old for Iris. Consider upgrading to kernel v4.16.`

Looks like most CBs released in the past couple of years are running v4.4 or v4.14, some are running v4.19 or v5.4 (e.g. @DennisLfromGA 's machine)   
https://www.chromium.org/chromium-os/developer-information-for-chrome-os-devices

@yoiammaac , what kernel is your's running?

**Comment by yoiammaac** [comment]
> > @yoiammaac,
> > ```
> > $ edit-chroot -al
> > ----------------------------------------------------------------------
> > chroot name: bionic
> > crouton: version 1-20210709170820~master:14ca6bc7
> > release: bionic
> > architecture: amd64
> > xmethod: xiwi
> > targets: xiwi,xorg,cli-extra,keyboard,touch,gnome-desktop,lxde-desktop,chrome-beta
> > host: version 13982.39.0 (Official Build) beta-channel eve 
> > kernel: Linux localhost 5.4.119-14931-g1ec34905b11c #1 SMP PREEMPT Wed Jun 30 20:51:02 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
> > freon: yes
> > ----------------------------------------------------------------------
> > chroot name: focal
> > crouton: version 1-20210709170820~master:14ca6bc7
> > release: focal
> > architecture: amd64
> > xmethod: xiwi
> > targets: xiwi,xorg,cli-extra,keyboard,touch,gnome-desktop,lxde-desktop,chrome-beta
> > host: version 13982.39.0 (Official Build) beta-channel eve 
> > kernel: Linux localhost 5.4.119-14931-g1ec34905b11c #1 SMP PREEMPT Wed Jun 30 20:51:02 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
> > freon: yes
> > ----------------------------------------------------------------------
> > chroot name: xenial
> > crouton: version 1-20210709170820~master:14ca6bc7
> > release: xenial
> > architecture: amd64
> > xmethod: xiwi
> > targets: xiwi,xorg,gnome-desktop,lxde-desktop,unity-desktop,xfce-desktop,cli-extra,keyboard,touch,chrome-beta
> > host: version 13982.39.0 (Official Build) beta-channel eve 
> > kernel: Linux localhost 5.4.119-14931-g1ec34905b11c #1 SMP PREEMPT Wed Jun 30 20:51:02 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
> > freon: yes
> > ----------------------------------------------------------------------```
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I have 3 chroots, including focal, all running and working fine with the latest crouton update/patches.
> > _Hope this helps,_
> > -**DennisLfromGA**
> 
> I'm curious if the kernel is a first order issue here.
> 
> I'm struggling to get a functioning install of focal xfce.
> 
> The first error message I get is
> 
> `error: Kernel is too old for Iris. Consider upgrading to kernel v4.16.`
> 
> Looks like most CBs released in the past couple of years are running v4.4 or v4.14, some are running v4.19 or v5.4 (e.g. @DennisLfromGA 's machine)
> https://www.chromium.org/chromium-os/developer-information-for-chrome-os-devices
> 
> @yoiammaac , what kernel is your's running?
kernel: Linux localhost 4.4.254-37689-g90cbe4d754f7 #1 SMP PREEMPT Sun Jun 27 23:02:44 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux

**Comment by Dozer257** [comment]
> UPDATE: FINALLY (took me literally 3 minutes to get audio setup using that guide: My complete tutorial:
> Install focal Fossa with this command: sudo crouton -r focal -t xorg,audio,core,gtk-extra,x11,xfce. Wait for crouton to do its thing. Create a username and pass, then sudo enter-chroot and enter the following commands:
> 
> sudo apt-get remove --purge pulseaudio alsa-base
> sudo apt-get install alsa-base
> sudo usermod -a -G hwaudio "$USER" This command adds you to the hardware audio group.
> Exit the chroot, then on Chrome-OS, enter the following command:
> sudo initctl stop cras
> This stops cras (the default audio server crouton uses, which doesnt work so is useless).
> Then sudo enter-chroot again and enter the command:
> cat /proc/asound/cards.
> This will list off your audio cards on your chromebook, my cards name is chtmax98090 but yours may be different. The name of your card will be the second line after "1 [" e.g. mine was 1 [chtmax98090
> Now enter the command nano ~/.asoundrc and paste the following:
> 
> https://hastebin.com/usiqejifuv.shell
> 
> I had to put it in hastebin because github was doing weird formatting stuff
> 
> Nano may not be installed so install it using:
> sudo apt install nano -y
> while were at it install pulseaudio again:
> sudo apt install pulseaudio -y
> (the -y flag skips the yes or no confirmation with apt) Once you've pasted that into ~/.asoundrc, use the keybind: ctrl+o to save and press enter, then ctrl+x to exit. Next install gnome control center if not already installed using:
> sudo apt install gnome-control-center
> Then when its installed exit the chroot and start xfce using:
> sudo startxfce4
> Go to your settings and click session and startup (or something like that)
> then click applications startup at the top and click add at the bottom.
> We need to execute two commands at startup: You can name these whatever you want with a description of whatever you want but for the command we need these commands:
> pulseaudio --start
> then click ok
> now click add one more time, again the title and description can be whatever you want but the command must be:
> xfce4-panel
> The pulseaudio start command starts the pulseaudio server and the panel command insures the panel starts everytime. (otherwise you may have a hard time using your applications)
> Now in order for audio to start working, exit the chroot and start xfce again. Now just customize and your done!

It worked for me, however, the only issue I have is the fact that when I use a headset that uses the audio jack, I can't use it's microphone because it hasn't been detected. But i have a usb headset that has a mic that gets detected just fine, is there a fix for this?

**Comment by yoiammaac** [comment]
@DennisLfromGA  @dnschneid 
I installed bionic with [this](https://github.com/dnschneid/crouton/wiki/3-Steps-to-Install-Bionic-Beaver) and bionic works so fine(much better than xenial), its as if bionic is supported but, xenial isnt. Focal works fine(almost as good as bionic), though with no audio. 
Please bring audio at least for focal, if it cant be termed supported.

**Comment by MrHallway0** [comment]
> @DennisLfromGA @dnschneid
> I installed bionic with [this](https://github.com/dnschneid/crouton/wiki/3-Steps-to-Install-Bionic-Beaver) and bionic works so fine(much better than xenial), its as if bionic is supported but, xenial isnt. Focal works fine(almost as good as bionic), though with no audio.
> Please bring audio at least for focal, if it cant be termed supported.

1. Please don't mention people if they don't have much to do with it. 
2. This is about Focal Fossa, not Bionic and Xenial. 
3. @TheRealFanjin solved the audio issues with this: 

>  I think I solved it by doing `pulseaudio -D` or `pulseaudio --start`, I forgot which one it was.

**Comment by yoiammaac** [comment]
> > @DennisLfromGA @dnschneid
> > I installed bionic with [this](https://github.com/dnschneid/crouton/wiki/3-Steps-to-Install-Bionic-Beaver) and bionic works so fine(much better than xenial), its as if bionic is supported but, xenial isnt. Focal works fine(almost as good as bionic), though with no audio.
> > Please bring audio at least for focal, if it cant be termed supported.
> 
> 1. Please don't mention people if they don't have much to do with it.
> 2. This is about Focal Fossa, not Bionic and Xenial.
> 3. @TheRealFanjin solved the audio issues with this:
> 
> > I think I solved it by doing `pulseaudio -D` or `pulseaudio --start`, I forgot which one it was.

Sorry

**Comment by adamallgood** [comment]
> @yoiammaac,
> 
> ```
> $ edit-chroot -al
> ----------------------------------------------------------------------
> chroot name: bionic
> crouton: version 1-20210709170820~master:14ca6bc7
> release: bionic
> architecture: amd64
> xmethod: xiwi
> targets: xiwi,xorg,cli-extra,keyboard,touch,gnome-desktop,lxde-desktop,chrome-beta
> host: version 13982.39.0 (Official Build) beta-channel eve 
> kernel: Linux localhost 5.4.119-14931-g1ec34905b11c #1 SMP PREEMPT Wed Jun 30 20:51:02 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
> freon: yes
> ----------------------------------------------------------------------
> chroot name: focal
> crouton: version 1-20210709170820~master:14ca6bc7
> release: focal
> architecture: amd64
> xmethod: xiwi
> targets: xiwi,xorg,cli-extra,keyboard,touch,gnome-desktop,lxde-desktop,chrome-beta
> host: version 13982.39.0 (Official Build) beta-channel eve 
> kernel: Linux localhost 5.4.119-14931-g1ec34905b11c #1 SMP PREEMPT Wed Jun 30 20:51:02 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
> freon: yes
> ----------------------------------------------------------------------
> chroot name: xenial
> crouton: version 1-20210709170820~master:14ca6bc7
> release: xenial
> architecture: amd64
> xmethod: xiwi
> targets: xiwi,xorg,gnome-desktop,lxde-desktop,unity-desktop,xfce-desktop,cli-extra,keyboard,touch,chrome-beta
> host: version 13982.39.0 (Official Build) beta-channel eve 
> kernel: Linux localhost 5.4.119-14931-g1ec34905b11c #1 SMP PREEMPT Wed Jun 30 20:51:02 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
> freon: yes
> ----------------------------------------------------------------------```
> ```
> 
> I have 3 chroots, including focal, all running and working fine with the latest crouton update/patches.
> 
> _Hope this helps,_
> -**DennisLfromGA**

@DennisLfromGA did you have to do anything special to get the gnome desktop to open with focal? I tried `sudo crouton -r focal -t keyboard,audio,gnome-desktop -n focalgnome` yesterday, and the chroot was created successfully (yay!), but when I tried startgnome, I got a blank screen with the mouse pointer, then a crash and the chroot unmounted. XFCE works fine.

**Comment by DennisLfromGA** [comment]
@adamallgood,

I just tested the 'gnome' desktop target on 'focal' again today - it failed.

The 'gnome' desktop target worked on both of my 'bionic' & 'xenial' chroots with both 'xorg' & 'xiwi' methods tho.

I saw [an older issue](https://github.com/dnschneid/crouton/issues/4163#issuecomment-562994132) that removed the `~/.confg` folder and tried it, it worked once but failed after that.
I updated the chroot (`sudo crouton -n focal -u`) and it worked just once again but failed after that.

I get the below error:
> /usr/bin/xinit: connection to X server lost  

but I haven't been able to determine exactly what's wrong, sorry.

-**DennisLfromGA**

**Comment by X4X5** [comment]
As a rhythm gamer, I've found that using CRAS is more accurate for timing than PulseAudio is. With that being said, do we really need to install PulseAudio during installation?

**Comment by sanrobin** [comment]
> UPDATE: FINALLY (took me literally 3 minutes to get audio setup using that guide: My complete tutorial:
> Install focal Fossa with this command: sudo crouton -r focal -t xorg,audio,core,gtk-extra,x11,xfce. Wait for crouton to do its thing. Create a username and pass, then sudo enter-chroot and enter the following commands:
> 
> sudo apt-get remove --purge pulseaudio alsa-base
> sudo apt-get install alsa-base
> sudo usermod -a -G hwaudio "$USER" This command adds you to the hardware audio group.
> Exit the chroot, then on Chrome-OS, enter the following command:
> sudo initctl stop cras
> This stops cras (the default audio server crouton uses, which doesnt work so is useless).
> Then sudo enter-chroot again and enter the command:
> cat /proc/asound/cards.
> This will list off your audio cards on your chromebook, my cards name is chtmax98090 but yours may be different. The name of your card will be the second line after "1 [" e.g. mine was 1 [chtmax98090
> Now enter the command nano ~/.asoundrc and paste the following:
> 
> https://hastebin.com/usiqejifuv.shell
> 
> I had to put it in hastebin because github was doing weird formatting stuff
> 
> Nano may not be installed so install it using:
> sudo apt install nano -y
> while were at it install pulseaudio again:
> sudo apt install pulseaudio -y
> (the -y flag skips the yes or no confirmation with apt) Once you've pasted that into ~/.asoundrc, use the keybind: ctrl+o to save and press enter, then ctrl+x to exit. Next install gnome control center if not already installed using:
> sudo apt install gnome-control-center
> Then when its installed exit the chroot and start xfce using:
> sudo startxfce4
> Go to your settings and click session and startup (or something like that)
> then click applications startup at the top and click add at the bottom.
> We need to execute two commands at startup: You can name these whatever you want with a description of whatever you want but for the command we need these commands:
> pulseaudio --start
> then click ok
> now click add one more time, again the title and description can be whatever you want but the command must be:
> xfce4-panel
> The pulseaudio start command starts the pulseaudio server and the panel command insures the panel starts everytime. (otherwise you may have a hard time using your applications)
> Now in order for audio to start working, exit the chroot and start xfce again. Now just customize and your done!

Um, The hastebin link is broken for me. can you give another link?

**Comment by Putu-Justine** [comment]
hey anyone want to test my fix?

technically not my fix since this is a mix of other audio fix for crouton laying around on the internet

thx for whatever guy that recover the original hastebin and posted it to pastebin!

tested on ChromeOS (rammus) with brunch, on Asus_X407UF (Intel i7-8550U)

[Pastebin](https://pastebin.com/tsjdPiYU)

**Comment by krome-3k** [comment]
Please make this work on ARM64 chromebooks.

**Comment by BetaPictoris** [comment]
> Please make this work on ARM64 chromebooks.

Please stay on topic if you are looking for Unity support on ARM it is not going to happen! 
Ubuntu *can* still be ran on ARM64 you just need to use a different desktop, considering Unity is discontinued it doesn't matter much anyway.

**Comment by krome-3k** [comment]
No i am not looking for unity desktop.. i use xfce and for arm64 chromebooks crouton only has a stable xenial. I would very much like to use a stable fossa with xfce on arm64 chromebooks with sound and everything else working. I can offer to test it on my device to any developer who is involved with fossa on crouton. Shoutout to all involved devs.

**Comment by Litonewo** [comment]
What desktops currently work with Focal?

**Comment by krome-3k** [comment]
I will check and get back to you

**Comment by BetaPictoris** [comment]
> No i am not looking for unity desktop.. i use xfce and for arm64 chromebooks crouton only has a stable xenial. I would very much like to use a stable fossa with xfce on arm64 chromebooks with sound and everything else working. I can offer to test it on my device to any developer who is involved with fossa on crouton. Shoutout to all involved devs.

Please read the [README's](https://github.com/dnschneid/crouton/issues/4265#issuecomment-921101671) documentation on other releases. You ***must*** specify the `-r` option!

**Comment by BetaPictoris** [comment]
> What desktops currently work with Focal?

I have, personally, seen good success with XFCE...

**Comment by sanrobin** [comment]
> I have, personally, seen good success with XFCE...

Currently trying to install e17 in focal.

**Comment by sanrobin** [comment]
> I am currently trying to install e17 in focal. Hope it works.

nah i didnt work.

**Comment by krome-3k** [comment]
> > No i am not looking for unity desktop.. i use xfce and for arm64 chromebooks crouton only has a stable xenial. I would very much like to use a stable fossa with xfce on arm64 chromebooks with sound and everything else working. I can offer to test it on my device to any developer who is involved with fossa on crouton. Shoutout to all involved devs.
> 
> Please read the [README's](https://github.com/dnschneid/crouton/issues/4265#issuecomment-921101671) documentation on other releases. You ***must*** specify the `-r` option! 
> 
> 

Yes i know

**Comment by krome-3k** [comment]
> > > No i am not looking for unity desktop.. i use xfce and for arm64 chromebooks crouton only has a stable xenial. I would very much like to use a stable fossa with xfce on arm64 chromebooks with sound and everything else working. I can offer to test it on my device to any developer who is involved with fossa on crouton. Shoutout to all involved devs.
> > 
> > Please read the [README's](https://github.com/dnschneid/crouton/issues/4265#issuecomment-921101671) documentation on other releases. You ***must*** specify the `-r` option! 
> > 
> > 
> 
> Yes i know.. stop stating the obvious

**Comment by BetaPictoris** [comment]
> > > No i am not looking for unity desktop.. i use xfce and for arm64 chromebooks crouton only has a stable xenial. I would very much like to use a stable fossa with xfce on arm64 chromebooks with sound and everything else working. I can offer to test it on my device to any developer who is involved with fossa on crouton. Shoutout to all involved devs.
> > 
> > 
> > Please read the [README's](https://github.com/dnschneid/crouton/issues/4265#issuecomment-921101671) documentation on other releases. You _**must**_ specify the `-r` option!
> 
> Yes i know.. stop stating the obvious

Okay, then I don't know what you want me to say. I have a Focal XFCE chroot setup, on `arch64`.

**Comment by BetaPictoris** [comment]
> > I am currently trying to install e17 in focal. Hope it works.
> 
> nah i didnt work.

Not surprised. E17 was released somewhere around 2012... I wouldn't even think you could get it to work, through a PPA, on a normal Focal device!

**Comment by krome-3k** [comment]
> > > > No i am not looking for unity desktop.. i use xfce and for arm64 chromebooks crouton only has a stable xenial. I would very much like to use a stable fossa with xfce on arm64 chromebooks with sound and everything else working. I can offer to test it on my device to any developer who is involved with fossa on crouton. Shoutout to all involved devs.
> > > 
> > > 
> > > Please read the [README's](https://github.com/dnschneid/crouton/issues/4265#issuecomment-921101671) documentation on other releases. You _**must**_ specify the `-r` option!
> > 
> > Yes i know.. stop stating the obvious
> 
> Okay, then I don't know what you want me to say. I have a Focal XFCE chroot setup, on `arch64`.

Is audio working??

**Comment by ZZZXXX0110** [comment]
My audio is working on Focal with xfce4 and with Sommelier with the fix listed here. But I recently noticed that launching any GUI app or the xfce4 desktop in Crouton would break HDMI audio output for the entire Chrome OS including any apps running on Crouton, HDMI audio won't work for Android apps or Chrome OS native webapps, although my device's built in speaker and Bluetooth audio are not affected. The only way to fix it seems to do a complete reboot. 

I made a new issue post on this with more information: https://github.com/dnschneid/crouton/issues/4605

**Comment by jefbal99** [comment]
> UPDATE: FINALLY (took me literally 3 minutes to get audio setup using that guide: My complete tutorial: Install focal Fossa with this command: sudo crouton -r focal -t xorg,audio,core,gtk-extra,x11,xfce. Wait for crouton to do its thing. Create a username and pass, then sudo enter-chroot and enter the following commands:
> 
> sudo apt-get remove --purge pulseaudio alsa-base sudo apt-get install alsa-base sudo usermod -a -G hwaudio "$USER" This command adds you to the hardware audio group. Exit the chroot, then on Chrome-OS, enter the following command: sudo initctl stop cras This stops cras (the default audio server crouton uses, which doesnt work so is useless). Then sudo enter-chroot again and enter the command: cat /proc/asound/cards. This will list off your audio cards on your chromebook, my cards name is chtmax98090 but yours may be different. The name of your card will be the second line after "1 [" e.g. mine was 1 [chtmax98090 Now enter the command nano ~/.asoundrc and paste the following:
> 
> https://hastebin.com/usiqejifuv.shell
> 
> I had to put it in hastebin because github was doing weird formatting stuff
> 
> Nano may not be installed so install it using: sudo apt install nano -y while were at it install pulseaudio again: sudo apt install pulseaudio -y (the -y flag skips the yes or no confirmation with apt) Once you've pasted that into ~/.asoundrc, use the keybind: ctrl+o to save and press enter, then ctrl+x to exit. Next install gnome control center if not already installed using: sudo apt install gnome-control-center Then when its installed exit the chroot and start xfce using: sudo startxfce4 Go to your settings and click session and startup (or something like that) then click applications startup at the top and click add at the bottom. We need to execute two commands at startup: You can name these whatever you want with a description of whatever you want but for the command we need these commands: pulseaudio --start then click ok now click add one more time, again the title and description can be whatever you want but the command must be: xfce4-panel The pulseaudio start command starts the pulseaudio server and the panel command insures the panel starts everytime. (otherwise you may have a hard time using your applications) Now in order for audio to start working, exit the chroot and start xfce again. Now just customize and your done!

The hastebin script for the audio is gone as hastebin was bought out, can it be reposted elsewhere?

**Comment by ShipmasterKyle** [comment]
For those of you having issues with the downloads I'll qoute kapilhp's soulution.
> The "quick and dirty" fix for this is to comment out the line revoking the key during `su -l`. The relevant line in `/etc/pam.d/su-l` should be commented out as below.
> 
> ```
> # session		optional	pam_keyinit.so force revoke
> ```
> 
> The point is that Debian/Ubuntu now revokes the keys when doing `su -l` by default. However, `crouton` activates the chroot via `su -l`. Perhaps `crouton` should be using some other method (`schroot` or other?) to enter the chroot.

**Comment by DennisLfromGA** [comment]
Since crouton is now in 'maintenance only' status, Focal Fossa will not be tested and made the default release.

For more details see this important PSA:
- crouton is now maintenance-only #4654
